src/SchemaParser.f90

```diff
@@ -770,67 +791,69 @@ SUBROUTINE clear_SchBlk(this)
 !>                   considered
 !>
-    SUBROUTINE parse_SchBlk(this,inputFile,paramList,ioccurBlk)
-      CHARACTER(LEN=*),PARAMETER :: myName='parse_SchBlk'
-      CLASS(SchemaBlockType),INTENT(INOUT) :: this
-      TYPE(InputFileType),INTENT(INOUT) :: inputFile
-      TYPE(ParamType),INTENT(INOUT) :: paramList
-      INTEGER(SIK),INTENT(IN) :: ioccurBlk
-
-      INTEGER(SIK) :: nerr
-      INTEGER(SIK) :: icard,ioccur,startLine,stopLine,startField,stopField
-      TYPE(StringType) :: pListPath
-      
-      nerr=eSchemaParser%getCounter(EXCEPTION_ERROR)
-
-      startLine=this%startLine(ioccurBlk); stopLine=this%stopLine(ioccurBlk);
-      startField=this%startField(ioccurBlk); stopField=this%stopField(ioccurBlk);
-      pListPath=''
-      CALL this%addPLPath(pListPath,ioccurBlk)
-
-      !Parse each Card
-      DO icard=1,SIZE(this%card)
-        CALL this%card(icard)%countOccurrences(inputFile,startLine,stopLine,startField,stopField)
-        IF(.NOT.this%card(icard)%nOccurrencesIsValid()) RETURN
-        CALL this%card(icard)%determineExtentsWithinTextFile(inputFile,this%card,startLine,stopLine,startField,stopField)
-        DO ioccur=1,this%card(icard)%nOccurrences
-          CALL this%card(icard)%parse(inputFile,paramList,ioccur,pListPath)
-        ENDDO
-      ENDDO
-
-      IF(eSchemaParser%getCounter(EXCEPTION_ERROR) > nerr) THEN
-        CALL eSchemaParser%raiseError(modName//'::'//myName//' - Error parsing block '//TRIM(this%name)//'!')
-      ENDIF
-    ENDSUBROUTINE parse_SchBlk
+SUBROUTINE parse_SchBlk(this,inputFile,paramList,ioccurBlk)
+  CHARACTER(LEN=*),PARAMETER :: myName='parse_SchBlk'
+  CLASS(SchemaBlock),INTENT(INOUT) :: this
+  TYPE(InputFileType),INTENT(INOUT) :: inputFile
+  TYPE(ParamType),INTENT(INOUT) :: paramList
+  INTEGER(SIK),INTENT(IN) :: ioccurBlk
+
+  INTEGER(SIK) :: nerr
+  INTEGER(SIK) :: icard,ioccur,firstLine,lastLine,firstField,lastField
+  TYPE(StringType) :: pListPath
+  CHARACTER(LEN=1000) ioccurChr
+  
+  nerr=eSchemaParser%getCounter(EXCEPTION_ERROR)
+
+  firstLine=this%firstLine(ioccurBlk); lastLine=this%lastLine(ioccurBlk);
+  firstField=this%firstField(ioccurBlk); lastField=this%lastField(ioccurBlk);
+  pListPath=''
+  CALL this%addPLPath(pListPath,ioccurBlk)
+
+  !Parse each Card
+  DO icard=1,SIZE(this%card)
+    CALL this%card(icard)%countOccurrences(inputFile,firstLine,lastLine,firstField,lastField)
+    IF(.NOT.this%card(icard)%nOccurrencesIsValid()) RETURN
+    CALL this%card(icard)%determineExtentsWithinTextFile(inputFile,this%card,firstLine,lastLine,firstField,lastField)
+    DO ioccur=1,this%card(icard)%nOccurrences
+      CALL this%card(icard)%parse(inputFile,paramList,ioccur,pListPath)
+    ENDDO
+  ENDDO
+
+  IF(eSchemaParser%getCounter(EXCEPTION_ERROR) > nerr) THEN
+    WRITE(ioccurChr,'(I0)') ioccurBlk
+    CALL eSchemaParser%raiseError(modName//'::'//myName//' - Error parsing block '//TRIM(this%name)//' Instance '//TRIM(ioccurChr)//'!')
```

Comment:
So I literally just added this yesterday I think, but in `IO_Strings` or whatever the module is, there's now a `str` function that will take an integer and return the character version of that integer with no whitespace.  So you can get rid of line 824 and then replace `TRIM(ioccurChr)` with `str(ioccurBlk)` and get the same thing.  If you did this kind of writing to a string variable anywhere else, go ahead and replace those as well.

Note: If you haven't merged/rebased with master, you'll need that to get the `str` function.